File: Algorithms/Python/Container-With-Most-Water.py
class Solution(object):

    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        slow, fast = 0, len(height) - 1
        ans = 0
        while slow < fast:
            ans = max(ans, (fast - slow) * min(height[fast], height[slow]))
            if height[slow] < height[fast]:
                slow += 1
            elif height[slow] > height[fast]:
                fast -= 1
            else:
                slow += 1
                fast -= 1
        return ans

File: Algorithms/Java/Container With Most Water.java
public class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int Area = 0;
        while (left < right) {
            Area = Math.max(Area, Math.min(height[left], height[right])
                    * (right - left));
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return Area;       
    }
}